src/main/java/frc/robot/commands/shooter/FullAutoShooterAssembly.java

```diff
@@ -0,0 +1,323 @@
+    protected double currentMotorSpeed;
+    protected double currentIndexSpeed;
+    protected boolean isDrivenAndAligned;
+       
+    // PID values
+    protected static final double
+        ALIGN_P = 0,
+        ALIGN_I = 0,
+        ALIGN_D = 0,
+        MOVE_P = 0,
+        MOVE_I = 0,
+        MOVE_D = 0;
+
+    // devices
+    private Lemonlight limelight;
+
+    /**
+     * Command for running the shooter in full auto mode.
+     *
+     * @param shooter The shooter subsystem.
+     * @param conveyor The conveyor subsystem.
+     * @param drivetrain The drivetrain subsystem.
+     * @param limelight The limelight device.
+     */
+    public FullAutoShooterAssembly(Shooter shooter,
+        Conveyor conveyor,
+        Drivetrain drivetrain,
+        Lemonlight limelight) {
+
+        this.shooter = shooter;
+        this.conveyor = conveyor;
+        this.drivetrain = drivetrain;
+        this.limelight = limelight;
+
+        alignRightPID = new PIDController(ALIGN_P, ALIGN_I, ALIGN_D);
+        alignWrongPID = new PIDController(ALIGN_P, ALIGN_I, ALIGN_D);
+        movePID = new PIDController(MOVE_P, MOVE_I, MOVE_D);
+
+        // TODO - Set these, including the constants
+        alignRightPID.setTolerance(Shooter.TARGET_HORIZONTAL_ACCURACY, 1);
+        alignWrongPID.setTolerance(Shooter.TARGET_HORIZONTAL_ACCURACY, 1);
+        movePID.setTolerance(1, 1);
+        alignRightPID.setSetpoint(0);
+        alignWrongPID.setSetpoint(Shooter.TARGET_WRONG_COLOR_MISS);
+        movePID.setSetpoint(Drivetrain.IDEAL_SHOOTING_DISTANCE);
+
+        addRequirements(shooter, drivetrain, conveyor);
+    }
+
+    /**
+     * Returns the team color as a ConveyorState.
+     *
+     * @return color The team color as a ConveyorState.
+     */
+    public ConveyorState getTeamColor() {
+        if (DriverStation.getAlliance().toString() == "kRed") {
+            return ConveyorState.RED;
+        } else {
+            return ConveyorState.BLUE;
+        }
+    }
+
+    /**
+     * Returns the desired motor speed based on the distance from the target and the hood position.
+     *
+     * @param distance The distance from the target.
+     * @param hoodPos The hood position.
+     * @return motorSpeed The desired motor speed
+     */
+    public double solveMotorSpeed(double distance, boolean hoodPos) {
+        // Annoyingly, exponent notation requires importing a math library.
+        // So, for simplicity, we do not use it.
+        // TODO - Test the shooter and do a regression to find the right formula.
+        // Add higher order terms if necessary.
+        if (hoodPos) {
+            return 0 * distance * distance * distance
+                + 0 * distance * distance
+                + 0 * distance
+                + 0;
+        } else {
+            return 0 * distance * distance * distance
+                + 0 * distance * distance
+                + 0 * distance
+                + 0;
+        }
+    }
+
+    public void fire() {
```

Comment:
chage fire to use the sparks built in pid to turn the whele ~180 degrees instead of trying to set the power (add a minimum delay between fireing balls as well)